nawab.py

```diff
@@ -4,67 +4,100 @@
 
     dp = updater.dispatcher
     dp.add_handler(CommandHandler('start', bot.start))
-    dp.add_handler(CommandHandler("stop", bot.stop, pass_chat_data=True))
+    dp.add_handler(CommandHandler('stop', bot.stop, pass_chat_data=True))
     dp.add_handler(CommandHandler('help',bot.help))
     dp.add_handler(CallbackQueryHandler(bot.button))
     dp.add_error_handler(bot.error)
 
     # Start the Bot
-    updater.start_polling()
-
+    if isretweet == False:
+        updater.start_polling()
+    
     # Run the bot until the user presses Ctrl-C or the process receives SIGINT,
     # SIGTERM or SIGABRT
     updater.idle()
 
-
 def wrapper(func, args, res):
     res.append(func(*args))
 
 
 def main():
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument("-r", "--retweet", help="Retweet all tweets automatically, doesn't spawn a telegram bot",
+                        action='store_true', required=False)
+    parser.add_argument("-b", "--blacklist",type=list, required=False, help="Blacklist the given username")
+    parser.add_argument("-p", "--path", type=str, required=False,
+                        help="Path where the log files be stored. Note to create directory in that path beforehand.")
+    parser.add_argument('-d', '--default', action="store_const", const=30)
+    parser.add_argument('-V', '--verbose',action="store_const", const=20)
+    parser.add_argument('-s', '--silent', action="store_const", const=50)
+    parser.add_argument('-h', '--help', action='help', default=argparse.SUPPRESS,
+                        help='Show this help message and exit.')
+    parser.add_argument('-v', '--version', action='version',
+                        version='%(prog)s 1.0', help="Show program's version number and exit.")
+
+    args = vars(parser.parse_args())
+    ## set manual loglevels
+    if args['default']:
+        level = args['default']
+    elif args['verbose']:
+        level = args['verbose']
+    elif args['silent']:
+        level = args['silent']
+    else:
+        level = 10
+    names = args['blacklist']
+
     res = []
     data = pd.read_csv('data.csv')
     default_dir = '/var/log/nawab/'
+    ## blacklisting the names to current data.csv files
+    if names:
+        blist = pd.DataFrame({'Blacklist': names})
+        concats = pd.concat([data,blist])
+        concats.to_csv('data.csv', index=False)
+    ## if new path is passed then set new default path
+    if args['path']:
+        default_dir = args['path']
 
     u_id = pwd.getpwuid(os.getuid()).pw_name
 
-    if not os.path.exists(default_dir):
-        os.system(("sudo mkdir %s" % (default_dir)))
-
     ownership_command = "sudo chown %s: %s" % (u_id, default_dir)
     os.system(ownership_command)
-
+    
     #Initiate twitter bot
     thread = threading.Thread(target=wrapper, args=(
-        twitter_bot_run, (data, default_dir, ), res))
+        twitter_bot_run, (data, default_dir,level,args['retweet'], ), res))
```

Comment:
why pass that always? what is args.retweet isn't set at the first place? Wouldn't that produce as missing positional argument error?